tests/test_catalogs.py

```diff
@@ -1,183 +1,190 @@
     g = Graph()
-    g.parse(data=resp.text, format='json-ld')
+    g.parse(data=resp.text, format="json-ld")
     assert 0 < len(g)
 
 
-def _dump_turtle_sorted(g):
-    for l in sorted(g.serialize(format='turtle').splitlines()):
-        if l:
-            print(l.decode())
+@pytest.mark.contract
+def test_isomorphic(http_service: Any) -> None:
+    """Should return a body with a graph that is isomorphic to correct graph."""
+    url = f"{http_service}/catalogs/1"
+    headers = {"Accept": "text/turtle"}
+    resp = requests.get(url, headers=headers)
+    g1 = Graph().parse(data=resp.text, format="turtle")
 
+    g2 = Graph().parse("tests/catalog_1.ttl", format="turtle")
 
-def test_isomorphic():
-    "GET request to url returns a 200"
-    url = f"{HOST_URL}/catalogs/1"
-    headers = {'Accept': 'text/turtle'}
-    resp = requests.get(url, headers=headers)
-    assert 200 == resp.status_code
-    assert 0 < len(resp.content)
-    assert 'text/turtle; charset=utf-8' == resp.headers['Content-Type']
-    g1 = Graph()
-    g1.parse(data=resp.text, format='turtle')
-    assert 0 < len(g1)
-    g2 = Graph().parse("tests/catalog_1.ttl", format='turtle')
+    _isomorphic = isomorphic(g1, g2)
+    if not _isomorphic:
+        _dump_diff(g1, g2)
+        pass
+    assert _isomorphic
+
+
+# - BAD CASES:
+@pytest.mark.contract
+def test_not_found_gives_404(http_service: Any) -> None:
+    """Should return 404."""
+    url = f"{http_service}/catalogs/ID_NOT_IN_DB"
+    resp = requests.get(url)
+    assert 404 == resp.status_code
+
+
+# ---------------------------------------------------------------------- #
+# Utils for displaying debug information
+
+
+def _dump_diff(g1: Graph, g2: Graph) -> None:
     in_both, in_first, in_second = graph_diff(g1, g2)
     print("\nin both:")
-    _dump_turtle_sorted(in_both)
+    _dump_turtle(in_both)
     print("\nin first:")
-    _dump_turtle_sorted(in_first)
+    _dump_turtle(in_first)
     print("\nin second:")
-    _dump_turtle_sorted(in_second)
-    assert isomorphic(g1, g2)
+    _dump_turtle(in_second)
 
 
-def test_not_isomorphic():
-    "GET request to url returns a 200"
-    url = f"{HOST_URL}/catalogs/1"
-    headers = {'Accept': 'text/turtle'}
-    resp = requests.get(url, headers=headers)
-    assert 200 == resp.status_code
-    assert 0 < len(resp.content)
-    assert 'text/turtle; charset=utf-8' == resp.headers['Content-Type']
-    g1 = Graph()
-    g1.parse(data=resp.text, format='turtle')
-    assert 0 < len(g1)
-    g2 = Graph().parse("tests/catalog_2.ttl", format='turtle')
-    assert not isomorphic(g1, g2)
-
-
-# BAD CASES:
-def test_not_found_gives_404():
-    "GET request to url returns a 404"
-    url = f"{HOST_URL}/catalogs/ID_NOT_IN_DB"
-    resp = requests.get(url)
-    assert 404 == resp.status_code
+def _dump_turtle(g: Graph) -> None:
+    for l in g.serialize(format="turtle").splitlines():
+        if l:
+            print(l.decode())
```

Comment:
Nice! Hadde vært ekstra stas med test på at innholdet er som forventet (at den sender ut den katalogen du forventer altså) i catalogoue_by_id testene blir det enda bedre.

Reply:
(samme for test_dataservices)

Reply:
Litt usikker på kva du meiner her. Funksjonen du viser til, er kun ein "utility" for å printe til stdout dersom testen feiler. Viss eg forstod kommentaren din riktig, utfører følgande test det du er ute etter (https://github.com/Informasjonsforvaltning/dataservice-publisher/blob/499fcf223ff0e4a9cc8fda450a9fb224a7fe55d7/tests/test_catalogs.py#L147): 
```
@pytest.mark.contract
def test_isomorphic(http_service: Any) -> None:
    """Should return a body with a graph that is isomorphic to correct graph."""
    url = f"{http_service}/catalogs/1"
    headers = {"Accept": "text/turtle"}
    resp = requests.get(url, headers=headers)
    g1 = Graph().parse(data=resp.text, format="turtle")

    g2 = Graph().parse("tests/catalog_1.ttl", format="turtle")

    _isomorphic = isomorphic(g1, g2)
    if not _isomorphic:
        _dump_diff(g1, g2)
        pass
    assert _isomorphic
```
Tilsvarande i test_dataservices.py